File: app/src/main/java/org/fossasia/susi/ai/rest/model/RelatedTopics.java
package org.fossasia.susi.ai.rest.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by mayank on 09-12-2016.
 */

public class RelatedTopics {

    @SerializedName("FirstURL")
    @Expose
    private String des;

    @SerializedName("Text")
    @Expose
    private String text;

    @SerializedName("Icon")
    @Expose
    private WebIcon icon;


    public RelatedTopics(String des, String text, WebIcon icon) {
        this.des = des;
        this.text = text;
        this.icon = icon;
    }

    public String getDes() {
        return des;
    }

    public void setDes(String des) {
        this.des = des;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public WebIcon getIcon() {
        return icon;
    }

    public void setIcon(WebIcon icon) {
        this.icon = icon;
    }
}

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/others/RelatedTopics.kt
package org.fossasia.susi.ai.rest.responses.others

import com.google.gson.annotations.SerializedName

/**
 * <h1>Kotlin Data class to parse related topics object in retrofit response from websearch client.</h1>
 *
 * Created by mayank on 09-12-2016.
 */

data class RelatedTopics(
    @SerializedName("FirstURL")
    var url: String? = null,
    @SerializedName("Text")
    var text: String? = null,
    @SerializedName("Icon")
    var icon: WebIcon? = null,
    @SerializedName("Result")
    var result: String? = null
)
